phoenix-core/src/main/java/org/apache/phoenix/schema/PTableImpl.java

```diff
@@ -1038,32 +1038,21 @@ public void setValue(PColumn column, byte[] byteValue) {
         @Override
         public void delete() {
             newMutations();
-            // we're using the Tephra column family delete marker here to prevent the translation 
-            // of deletes to puts by the Tephra's TransactionProcessor
-            if (PTableImpl.this.isTransactional()) {
-                Put put = new Put(key);
-                if (families.isEmpty()) {
-                    put.add(SchemaUtil.getEmptyColumnFamily(PTableImpl.this), TransactionFactory.getTransactionFactory().getTransactionContext().getFamilyDeleteMarker(), ts,
-                            HConstants.EMPTY_BYTE_ARRAY);
-                } else {
-                    for (PColumnFamily colFamily : families) {
-                        put.add(colFamily.getName().getBytes(), TransactionFactory.getTransactionFactory().getTransactionContext().getFamilyDeleteMarker(), ts,
-                                HConstants.EMPTY_BYTE_ARRAY);
-                    }
-                }
-                deleteRow = put;                
```

Comment:
Does the code in IndexMaintainer here need to change too (i.e. check for getFamilyDeleteMarker()), or is it fine?

    private DeleteType getDeleteTypeOrNull(Collection<? extends Cell> pendingUpdates, int nCFs) {
        int nDeleteCF = 0;
        int nDeleteVersionCF = 0;
        for (Cell kv : pendingUpdates) {
        	if (kv.getTypeByte() == KeyValue.Type.DeleteFamilyVersion.getCode()) {
                nDeleteVersionCF++;
            }
        	else if (kv.getTypeByte() == KeyValue.Type.DeleteFamily.getCode()
        			// Since we don't include the index rows in the change set for txn tables, we need to detect row deletes that have transformed by TransactionProcessor
        			|| (CellUtil.matchingQualifier(kv, TransactionFactory.getTransactionFactory().getTransactionContext().getFamilyDeleteMarker()) && CellUtil.matchingValue(kv, HConstants.EMPTY_BYTE_ARRAY))) {
        	    nDeleteCF++;
        	}
        }

Reply:
No, since instead of writing directly the family deletion marker we perform a regular delete operation using the transaction processor. The transaction processor writes this family deletion marker and in here we just check for its existence.

Reply:
The pendingUpdates represent what is about to be written (i.e. the current batch of mutations). I think we need to leave it, though, if we want an old client to work with a new server. It doesn't hurt anything as far as I can see.